cedar-testing/src/cedar_test_impl.rs

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Definition of a `CedarTestImplementation` trait that describes an
+//! implementation of Cedar to use during testing.
+
+pub use cedar_policy::frontend::is_authorized::InterfaceResponse;
+use cedar_policy_core::ast::{Expr, PolicySet, Request, Value};
+use cedar_policy_core::authorizer::Authorizer;
+use cedar_policy_core::entities::Entities;
+use cedar_policy_core::evaluator::Evaluator;
+use cedar_policy_core::extensions::Extensions;
+use cedar_policy_validator::{ValidationMode, Validator, ValidatorSchema};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::time::{Duration, Instant};
+
+/// Return type for `CedarTestImplementation` methods
+#[derive(Debug, Deserialize)]
+pub enum TestResult<T> {
+    /// The request succeeded
+    Success(T),
+    /// The request failed (e.g., due to a parse error)
+    Failure(String),
+}
+
+impl<T> TestResult<T> {
+    /// Get the underlying value of a `TestResult`.
+    /// # Panics
+    /// If the `TestResult` is a `Failure`.
+    /// PANIC SAFETY only used in testing code
+    #[allow(clippy::panic)]
+    pub fn expect(self, msg: &str) -> T {
+        match self {
+            Self::Success(t) => t,
+            Self::Failure(err) => panic!("{msg}: {err}"),
+        }
+    }
+}
+
+/// Simple wrapper around u128 to remind ourselves that timing info is in microseconds.
+#[derive(Debug, Deserialize)]
+pub struct Micros(pub u128);
```

Comment:
Would it make sense to just directly store the `Duration`?

Reply:
This same type is used by the Lean FFI, so it makes it easier to just store the u128.